app/models/log_event.rb

```diff
@@ -19,11 +19,23 @@ def self.search(start_date: nil, end_date: nil, events: [], query: nil)
       start_date: start_date, end_date: end_date, events: events, query: query).search
   end
 
-# The reason that we are doing this is because in some case we can't add a default value to a text column 
+# The reason that we are doing this is because in some case we can't add a default value to a text column
   def metadata
     self[:metadata] || {}
   end
 
+  def facility
+    if loggable.respond_to?(:facility)
+      loggable.facility
+    elsif loggable.respond_to?(:price_group_with_deleted)
+      loggable.price_group_with_deleted.facility
+    elsif loggable.respond_to?(:facilities)
```

Comment:
I'm not really sure I like all these special cases.

I feel like there are two better options: 

- Drop this column and deal with the facility in the `to_log_s` methods (See `UserRole` for an example) 
- Delegation. I could see `PriceGroupMember` `delegate :facility, to: :price_group_with_deleted` (or it might even work as `belongs_to :facility, through: :price_group_with_deleted`

Happy to talk through this.

Reply:
I don't like the special cases either.  Taking another look at the ticket, it looks like the request is to know the facility for price groups and journals, so I think we can simply add the delegation to `PriceGroupMember` for now.  

The models that use the special case branches are `User`, `AccountUser`, and the various `Account` subclasses. These are not mentioned in the ticket - do you think it's worth asking the client about the importance of knowing the facility for those?